vega_sim/scenario/common/agents.py

```diff
@@ -709,3 +712,206 @@ def finalise(self):
+            time_in_force_opts (dict, optional):
+                Dictionary of time in force options and probabilities.
+            mean (float, optional):
+                Mean of the log-normal distribution.
+            sigma (float, optional):
+                Standard deviation of the log-normal distribution.
+        """
+
+        super().__init__(wallet_name + str(tag), wallet_pass)
+
+        self.market_name = market_name
+        self.asset_name = asset_name
+        self.spread = spread
+        self.initial_asset_mint = initial_asset_mint
+        self.buy_volume = buy_volume
+        self.sell_volume = sell_volume
+        self.tag = tag
+        self.random_state = (
+            random_state if random_state is not None else np.random.RandomState()
+        )
+        self.submit_bias = submit_bias
+        self.cancel_bias = cancel_bias
+        self.side_opts = side_opts
+        self.time_in_force_opts = time_in_force_opts
+        self.mean = mean
+        self.sigma = sigma
+
+    def initialise(self, vega: VegaServiceNull):
+        """Initialise the agents wallet and mint the required market asset.
+
+        Args:
+            vega (VegaServiceNull):
+                Object running a locally-hosted Vega service.
+        """
+
+        super().initialise(vega=vega)
+
+        self.market_id = [
+            m.id
+            for m in self.vega.all_markets()
+            if m.tradable_instrument.instrument.name == self.market_name
+        ][0]
+        self.asset_id = self.vega.find_asset_id(symbol=self.asset_name)
+        self.vega.mint(
+            self.wallet_name,
+            asset=self.asset_id,
+            amount=self.initial_asset_mint,
+        )
+        self.vega.wait_fn(2)
+
+    def step(self, vega_state: VegaState):
+        """Random submits and cancels limit orders.
+
+        Args:
+            vega_state (VegaState):
+                Object describing the state of the network and the market.
+        """
+
+        if (
+            vega_state.market_state[self.market_id].trading_mode
+            == markets_protos.Market.TradingMode.TRADING_MODE_CONTINUOUS
+        ) and vega_state.market_state[
+            self.market_id
+        ].state == markets_protos.Market.State.STATE_ACTIVE:
+
+            if self.random_state.rand() <= self.submit_bias:
+                self._submit_order()
+
+            if self.random_state.rand() <= self.cancel_bias:
+                self._cancel_order(vega_state=vega_state)
+
+    def _submit_order(self):
+
+        best_bid_price, best_offer_price = self.vega.best_prices(
+            market_id=self.market_id
+        )
+
+        side = self.random_state.choice(
+            a=list(self.side_opts.keys()),
+            p=list(self.side_opts.values()),
+        )
+        time_in_force = self.random_state.choice(
+            a=list(self.time_in_force_opts.keys()),
+            p=list(self.time_in_force_opts.values()),
+        )
```

Comment:
I think in the case of GTT we'll need to actually pass a time it is good until.. somewhere. I haven't used GTT yet but assume it'll be obvious in the docs 👍